tests/php/sync/test_class.jetpack-sync-themes.php

```diff
@@ -85,33 +88,27 @@ public function test_theme_callable_syncs_theme_supports_data() {
 		$this->sender->do_sync();
 		$theme_supports = $this->server_replica_storage->get_callable( 'theme_support' );
 
-		$theme_features = array(
-			'post-thumbnails',
-			'post-formats',
-			'custom-header',
-			'custom-background',
-			'custom-logo',
-			'menus',
-			'automatic-feed-links',
-			'editor-style',
-			'widgets',
-			'html5',
-			'title-tag',
-			'jetpack-social-menu',
-			'jetpack-responsive-videos',
-			'infinite-scroll',
-			'site-logo',
-			'editor-color-palette',
-			'editor-gradient-presets',
-		);
-
-		foreach ( $theme_features as $theme_feature ) {
+		foreach ( Defaults::$default_theme_support_whitelist as $theme_feature ) {
 			$this->assertEquals(
 				current_theme_supports( $theme_feature ),
 				isset( $theme_supports[ $theme_feature ] ),
-				'Feature(s) not synced' . $theme_feature
+				'Default Feature(s) not synced ' . $theme_feature
 			);
 		}
+
+		if ( ! function_exists( 'get_registered_theme_features' ) ) {
+			// get_registered_theme_features got intorduced in WP 5.5 so any tests that use the previous version of WP fail.
+			return;
+		}
+
+		// Sync all registered theme features.
+		$registered_theme_features = array_keys( get_registered_theme_features() );
+		$not_synced_theme_features = array_diff( $registered_theme_features, Defaults::$default_theme_support_whitelist );
+		// We want to make sure we keep up with the latest theme_supports data that gets registered.
+		$this->assertTrue(
```

Comment:
@enej Having the check here may be problematic. 

IF new support is added it will cause all tests to fail. Meaning the first PR post will see the issue and be forced to address it. I don't think is appropriate. Yes I want to make sure the list stays up to-date but it definitely shouldn't restrict others from continued development.

Reply:
I do want tests to fail when we miss syncing things. 

I do agree that fixing the issue should probably be done in a different PR. 
I think the PR to fix things would probably be fairly simple:
1) Add another key to the `Defaults::$default_theme_support_whitelist` list 
2) Make sure that the data get escaped properly and synced on the .com side. 

Could we make the test as something that breaks but is not blocking?

Reply:
see note for fuller thoughts. But this failure will occur against WP master not tied to individual PRs but globaly that a change was made to core that we need to account for which is appropriate.